From 862215637d4ec8f71b46ea5eeea03b59f9c36f89 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Mon, 3 Oct 2022 11:47:55 +0200
Subject: [PATCH 003/529] ARM: dts: stm32: remove stm32mp157*-scmi.dtb from
 compilation

To ease STM32MP157 ST boards rebase, remove stm32mp157*-scmi.dtb from
compilation.
stm32mp157*-scmi.dts will be used (after being renamed into .dtsi) as
include in stm32mp157*.dts.
Due to inclusion in stm32mp157*.dts, there is a special case
regarding stm32mp157c-ev1-scmi.dts(i). Indeed, stm32mp15-scmi.dtsi
include must be also removed to avoid double inclusion (-ev1.dts
includes ed1.dts which includes stm32mp15-scmi.dtsi).
Also remove *-scmi compatible from documentation.

Change-Id: I6651606a2a2462ed8a0140502f2fd14f6c5a22f3
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
---
 Documentation/devicetree/bindings/arm/stm32/stm32.yaml   | 8 --------
 arch/arm/boot/dts/Makefile                               | 4 ----
 ...tm32mp157a-dk1-scmi.dts => stm32mp157a-dk1-scmi.dtsi} | 6 ------
 ...tm32mp157c-dk2-scmi.dts => stm32mp157c-dk2-scmi.dtsi} | 6 ------
 ...tm32mp157c-ed1-scmi.dts => stm32mp157c-ed1-scmi.dtsi} | 6 ------
 ...tm32mp157c-ev1-scmi.dts => stm32mp157c-ev1-scmi.dtsi} | 9 ---------
 6 files changed, 39 deletions(-)
 rename arch/arm/boot/dts/{stm32mp157a-dk1-scmi.dts => stm32mp157a-dk1-scmi.dtsi} (87%)
 rename arch/arm/boot/dts/{stm32mp157c-dk2-scmi.dts => stm32mp157c-dk2-scmi.dtsi} (88%)
 rename arch/arm/boot/dts/{stm32mp157c-ed1-scmi.dts => stm32mp157c-ed1-scmi.dtsi} (88%)
 rename arch/arm/boot/dts/{stm32mp157c-ev1-scmi.dts => stm32mp157c-ev1-scmi.dtsi} (84%)

--- a/Documentation/devicetree/bindings/arm/stm32/stm32.yaml
+++ b/Documentation/devicetree/bindings/arm/stm32/stm32.yaml
@@ -81,26 +81,18 @@ properties:
           - const: st,stm32mp157
 
       - items:
-          - const: st,stm32mp157a-dk1-scmi
           - const: st,stm32mp157a-dk1
           - const: st,stm32mp157
       - items:
-          - const: st,stm32mp157c-dk2-scmi
           - const: st,stm32mp157c-dk2
           - const: st,stm32mp157
       - items:
-          - const: st,stm32mp157c-ed1-scmi
           - const: st,stm32mp157c-ed1
           - const: st,stm32mp157
       - items:
           - const: st,stm32mp157c-ev1
           - const: st,stm32mp157c-ed1
           - const: st,stm32mp157
-      - items:
-          - const: st,stm32mp157c-ev1-scmi
-          - const: st,stm32mp157c-ev1
-          - const: st,stm32mp157c-ed1
-          - const: st,stm32mp157
 
       - description: DH STM32MP1 SoM based Boards
         items:
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1223,7 +1223,6 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157a-avenger96.dtb \
 	stm32mp157a-dhcor-avenger96.dtb \
 	stm32mp157a-dk1.dtb \
-	stm32mp157a-dk1-scmi.dtb \
 	stm32mp157a-iot-box.dtb \
 	stm32mp157a-microgea-stm32mp1-microdev2.0.dtb \
 	stm32mp157a-microgea-stm32mp1-microdev2.0-of7.dtb \
@@ -1234,12 +1233,9 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dhcom-pdk2.dtb \
 	stm32mp157c-dhcom-picoitx.dtb \
 	stm32mp157c-dk2.dtb \
-	stm32mp157c-dk2-scmi.dtb \
 	stm32mp157c-ed1.dtb \
-	stm32mp157c-ed1-scmi.dtb \
 	stm32mp157c-emsbc-argon.dtb \
 	stm32mp157c-ev1.dtb \
-	stm32mp157c-ev1-scmi.dtb \
 	stm32mp157c-lxa-mc1.dtb \
 	stm32mp157c-odyssey.dtb
 dtb-$(CONFIG_MACH_SUN4I) += \
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157a-dk1-scmi.dtsi
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+#include "stm32mp15-scmi.dtsi"
+
+/ {
+	reserved-memory {
+		optee@de000000 {
+			reg = <0xde000000 0x2000000>;
+			no-map;
+		};
+	};
+};
+
+&cpu0 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cpu1 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&dsi {
+	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+};
+
+&gpioz {
+	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
+};
+
+&hash1 {
+	clocks = <&scmi_clk CK_SCMI_HASH1>;
+	resets = <&scmi_reset RST_SCMI_HASH1>;
+};
+
+&i2c4 {
+	clocks = <&scmi_clk CK_SCMI_I2C4>;
+	resets = <&scmi_reset RST_SCMI_I2C4>;
+};
+
+&iwdg2 {
+	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
+};
+
+&mdma1 {
+	resets = <&scmi_reset RST_SCMI_MDMA>;
+};
+
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
+};
+
+&rcc {
+	compatible = "st,stm32mp1-rcc-secure", "syscon";
+	clock-names = "hse", "hsi", "csi", "lse", "lsi";
+	clocks = <&scmi_clk CK_SCMI_HSE>,
+		 <&scmi_clk CK_SCMI_HSI>,
+		 <&scmi_clk CK_SCMI_CSI>,
+		 <&scmi_clk CK_SCMI_LSE>,
+		 <&scmi_clk CK_SCMI_LSI>;
+};
+
+&rng1 {
+	clocks = <&scmi_clk CK_SCMI_RNG1>;
+	resets = <&scmi_reset RST_SCMI_RNG1>;
+};
+
+&rtc {
+	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
+};
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+#include "stm32mp15-scmi.dtsi"
+
+/ {
+	reserved-memory {
+		optee@de000000 {
+			reg = <0xde000000 0x2000000>;
+			no-map;
+		};
+	};
+};
+
+&cpu0 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cpu1 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cryp1 {
+	clocks = <&scmi_clk CK_SCMI_CRYP1>;
+	resets = <&scmi_reset RST_SCMI_CRYP1>;
+};
+
+&dsi {
+	phy-dsi-supply = <&scmi_reg18>;
+	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+};
+
+&gpioz {
+	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
+};
+
+&hash1 {
+	clocks = <&scmi_clk CK_SCMI_HASH1>;
+	resets = <&scmi_reset RST_SCMI_HASH1>;
+};
+
+&i2c4 {
+	clocks = <&scmi_clk CK_SCMI_I2C4>;
+	resets = <&scmi_reset RST_SCMI_I2C4>;
+};
+
+&iwdg2 {
+	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
+};
+
+&mdma1 {
+	resets = <&scmi_reset RST_SCMI_MDMA>;
+};
+
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
+};
+
+&rcc {
+	compatible = "st,stm32mp1-rcc-secure", "syscon";
+	clock-names = "hse", "hsi", "csi", "lse", "lsi";
+	clocks = <&scmi_clk CK_SCMI_HSE>,
+		 <&scmi_clk CK_SCMI_HSI>,
+		 <&scmi_clk CK_SCMI_CSI>,
+		 <&scmi_clk CK_SCMI_LSE>,
+		 <&scmi_clk CK_SCMI_LSI>;
+};
+
+&rng1 {
+	clocks = <&scmi_clk CK_SCMI_RNG1>;
+	resets = <&scmi_reset RST_SCMI_RNG1>;
+};
+
+&rtc {
+	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
+};
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157c-ed1-scmi.dtsi
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+#include "stm32mp15-scmi.dtsi"
+
+/ {
+	reserved-memory {
+		optee@fe000000 {
+			reg = <0xfe000000 0x2000000>;
+			no-map;
+		};
+	};
+};
+
+&cpu0 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cpu1 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cryp1 {
+	clocks = <&scmi_clk CK_SCMI_CRYP1>;
+	resets = <&scmi_reset RST_SCMI_CRYP1>;
+};
+
+&dsi {
+	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+};
+
+&gpioz {
+	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
+};
+
+&hash1 {
+	clocks = <&scmi_clk CK_SCMI_HASH1>;
+	resets = <&scmi_reset RST_SCMI_HASH1>;
+};
+
+&i2c4 {
+	clocks = <&scmi_clk CK_SCMI_I2C4>;
+	resets = <&scmi_reset RST_SCMI_I2C4>;
+};
+
+&iwdg2 {
+	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
+};
+
+&mdma1 {
+	resets = <&scmi_reset RST_SCMI_MDMA>;
+};
+
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
+};
+
+&rcc {
+	compatible = "st,stm32mp1-rcc-secure", "syscon";
+	clock-names = "hse", "hsi", "csi", "lse", "lsi";
+	clocks = <&scmi_clk CK_SCMI_HSE>,
+		 <&scmi_clk CK_SCMI_HSI>,
+		 <&scmi_clk CK_SCMI_CSI>,
+		 <&scmi_clk CK_SCMI_LSE>,
+		 <&scmi_clk CK_SCMI_LSI>;
+};
+
+&rng1 {
+	clocks = <&scmi_clk CK_SCMI_RNG1>;
+	resets = <&scmi_reset RST_SCMI_RNG1>;
+};
+
+&rtc {
+	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
+};
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/ {
+	reserved-memory {
+		optee@fe000000 {
+			reg = <0xfe000000 0x2000000>;
+			no-map;
+		};
+	};
+};
+
+&cpu0 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cpu1 {
+	clocks = <&scmi_clk CK_SCMI_MPU>;
+};
+
+&cryp1 {
+	clocks = <&scmi_clk CK_SCMI_CRYP1>;
+	resets = <&scmi_reset RST_SCMI_CRYP1>;
+};
+
+&dsi {
+	phy-dsi-supply = <&scmi_reg18>;
+	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+};
+
+&gpioz {
+	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
+};
+
+&hash1 {
+	clocks = <&scmi_clk CK_SCMI_HASH1>;
+	resets = <&scmi_reset RST_SCMI_HASH1>;
+};
+
+&i2c4 {
+	clocks = <&scmi_clk CK_SCMI_I2C4>;
+	resets = <&scmi_reset RST_SCMI_I2C4>;
+};
+
+&iwdg2 {
+	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
+};
+
+&m_can1 {
+	clocks = <&scmi_clk CK_SCMI_HSE>, <&rcc FDCAN_K>;
+};
+
+&mdma1 {
+	resets = <&scmi_reset RST_SCMI_MDMA>;
+};
+
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
+};
+
+&rcc {
+	compatible = "st,stm32mp1-rcc-secure", "syscon";
+	clock-names = "hse", "hsi", "csi", "lse", "lsi";
+	clocks = <&scmi_clk CK_SCMI_HSE>,
+		 <&scmi_clk CK_SCMI_HSI>,
+		 <&scmi_clk CK_SCMI_CSI>,
+		 <&scmi_clk CK_SCMI_LSE>,
+		 <&scmi_clk CK_SCMI_LSI>;
+};
+
+&rng1 {
+	clocks = <&scmi_clk CK_SCMI_RNG1>;
+	resets = <&scmi_reset RST_SCMI_RNG1>;
+};
+
+&rtc {
+	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
+};
--- a/arch/arm/boot/dts/stm32mp157c-ev1-scmi.dts
+++ /dev/null
@@ -1,89 +0,0 @@
-// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
-/*
- * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
- * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
- */
-
-/dts-v1/;
-
-#include "stm32mp157c-ev1.dts"
-#include "stm32mp15-scmi.dtsi"
-
-/ {
-	model = "STMicroelectronics STM32MP157C-EV1 SCMI eval daughter on eval mother";
-	compatible = "st,stm32mp157c-ev1-scmi", "st,stm32mp157c-ed1", "st,stm32mp157";
-
-	reserved-memory {
-		optee@fe000000 {
-			reg = <0xfe000000 0x2000000>;
-			no-map;
-		};
-	};
-};
-
-&cpu0 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cpu1 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cryp1 {
-	clocks = <&scmi_clk CK_SCMI_CRYP1>;
-	resets = <&scmi_reset RST_SCMI_CRYP1>;
-};
-
-&dsi {
-	phy-dsi-supply = <&scmi_reg18>;
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
-};
-
-&gpioz {
-	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
-};
-
-&hash1 {
-	clocks = <&scmi_clk CK_SCMI_HASH1>;
-	resets = <&scmi_reset RST_SCMI_HASH1>;
-};
-
-&i2c4 {
-	clocks = <&scmi_clk CK_SCMI_I2C4>;
-	resets = <&scmi_reset RST_SCMI_I2C4>;
-};
-
-&iwdg2 {
-	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
-};
-
-&m_can1 {
-	clocks = <&scmi_clk CK_SCMI_HSE>, <&rcc FDCAN_K>;
-};
-
-&mdma1 {
-	resets = <&scmi_reset RST_SCMI_MDMA>;
-};
-
-&mlahb {
-	resets = <&scmi_reset RST_SCMI_MCU>;
-};
-
-&rcc {
-	compatible = "st,stm32mp1-rcc-secure", "syscon";
-	clock-names = "hse", "hsi", "csi", "lse", "lsi";
-	clocks = <&scmi_clk CK_SCMI_HSE>,
-		 <&scmi_clk CK_SCMI_HSI>,
-		 <&scmi_clk CK_SCMI_CSI>,
-		 <&scmi_clk CK_SCMI_LSE>,
-		 <&scmi_clk CK_SCMI_LSI>;
-};
-
-&rng1 {
-	clocks = <&scmi_clk CK_SCMI_RNG1>;
-	resets = <&scmi_reset RST_SCMI_RNG1>;
-};
-
-&rtc {
-	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
-};
--- a/arch/arm/boot/dts/stm32mp157a-dk1-scmi.dts
+++ /dev/null
@@ -1,79 +0,0 @@
-// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
-/*
- * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
- * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
- */
-
-/dts-v1/;
-
-#include "stm32mp157a-dk1.dts"
-#include "stm32mp15-scmi.dtsi"
-
-/ {
-	model = "STMicroelectronics STM32MP157A-DK1 SCMI Discovery Board";
-	compatible = "st,stm32mp157a-dk1-scmi", "st,stm32mp157";
-
-	reserved-memory {
-		optee@de000000 {
-			reg = <0xde000000 0x2000000>;
-			no-map;
-		};
-	};
-};
-
-&cpu0 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cpu1 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
-};
-
-&gpioz {
-	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
-};
-
-&hash1 {
-	clocks = <&scmi_clk CK_SCMI_HASH1>;
-	resets = <&scmi_reset RST_SCMI_HASH1>;
-};
-
-&i2c4 {
-	clocks = <&scmi_clk CK_SCMI_I2C4>;
-	resets = <&scmi_reset RST_SCMI_I2C4>;
-};
-
-&iwdg2 {
-	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
-};
-
-&mdma1 {
-	resets = <&scmi_reset RST_SCMI_MDMA>;
-};
-
-&mlahb {
-	resets = <&scmi_reset RST_SCMI_MCU>;
-};
-
-&rcc {
-	compatible = "st,stm32mp1-rcc-secure", "syscon";
-	clock-names = "hse", "hsi", "csi", "lse", "lsi";
-	clocks = <&scmi_clk CK_SCMI_HSE>,
-		 <&scmi_clk CK_SCMI_HSI>,
-		 <&scmi_clk CK_SCMI_CSI>,
-		 <&scmi_clk CK_SCMI_LSE>,
-		 <&scmi_clk CK_SCMI_LSI>;
-};
-
-&rng1 {
-	clocks = <&scmi_clk CK_SCMI_RNG1>;
-	resets = <&scmi_reset RST_SCMI_RNG1>;
-};
-
-&rtc {
-	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
-};
--- a/arch/arm/boot/dts/stm32mp157c-dk2-scmi.dts
+++ /dev/null
@@ -1,85 +0,0 @@
-// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
-/*
- * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
- * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
- */
-
-/dts-v1/;
-
-#include "stm32mp157c-dk2.dts"
-#include "stm32mp15-scmi.dtsi"
-
-/ {
-	model = "STMicroelectronics STM32MP157C-DK2 SCMI Discovery Board";
-	compatible = "st,stm32mp157c-dk2-scmi", "st,stm32mp157";
-
-	reserved-memory {
-		optee@de000000 {
-			reg = <0xde000000 0x2000000>;
-			no-map;
-		};
-	};
-};
-
-&cpu0 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cpu1 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cryp1 {
-	clocks = <&scmi_clk CK_SCMI_CRYP1>;
-	resets = <&scmi_reset RST_SCMI_CRYP1>;
-};
-
-&dsi {
-	phy-dsi-supply = <&scmi_reg18>;
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
-};
-
-&gpioz {
-	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
-};
-
-&hash1 {
-	clocks = <&scmi_clk CK_SCMI_HASH1>;
-	resets = <&scmi_reset RST_SCMI_HASH1>;
-};
-
-&i2c4 {
-	clocks = <&scmi_clk CK_SCMI_I2C4>;
-	resets = <&scmi_reset RST_SCMI_I2C4>;
-};
-
-&iwdg2 {
-	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
-};
-
-&mdma1 {
-	resets = <&scmi_reset RST_SCMI_MDMA>;
-};
-
-&mlahb {
-	resets = <&scmi_reset RST_SCMI_MCU>;
-};
-
-&rcc {
-	compatible = "st,stm32mp1-rcc-secure", "syscon";
-	clock-names = "hse", "hsi", "csi", "lse", "lsi";
-	clocks = <&scmi_clk CK_SCMI_HSE>,
-		 <&scmi_clk CK_SCMI_HSI>,
-		 <&scmi_clk CK_SCMI_CSI>,
-		 <&scmi_clk CK_SCMI_LSE>,
-		 <&scmi_clk CK_SCMI_LSI>;
-};
-
-&rng1 {
-	clocks = <&scmi_clk CK_SCMI_RNG1>;
-	resets = <&scmi_reset RST_SCMI_RNG1>;
-};
-
-&rtc {
-	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
-};
--- a/arch/arm/boot/dts/stm32mp157c-ed1-scmi.dts
+++ /dev/null
@@ -1,84 +0,0 @@
-// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
-/*
- * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
- * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
- */
-
-/dts-v1/;
-
-#include "stm32mp157c-ed1.dts"
-#include "stm32mp15-scmi.dtsi"
-
-/ {
-	model = "STMicroelectronics STM32MP157C-ED1 SCMI eval daughter";
-	compatible = "st,stm32mp157c-ed1-scmi", "st,stm32mp157";
-
-	reserved-memory {
-		optee@fe000000 {
-			reg = <0xfe000000 0x2000000>;
-			no-map;
-		};
-	};
-};
-
-&cpu0 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cpu1 {
-	clocks = <&scmi_clk CK_SCMI_MPU>;
-};
-
-&cryp1 {
-	clocks = <&scmi_clk CK_SCMI_CRYP1>;
-	resets = <&scmi_reset RST_SCMI_CRYP1>;
-};
-
-&dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
-};
-
-&gpioz {
-	clocks = <&scmi_clk CK_SCMI_GPIOZ>;
-};
-
-&hash1 {
-	clocks = <&scmi_clk CK_SCMI_HASH1>;
-	resets = <&scmi_reset RST_SCMI_HASH1>;
-};
-
-&i2c4 {
-	clocks = <&scmi_clk CK_SCMI_I2C4>;
-	resets = <&scmi_reset RST_SCMI_I2C4>;
-};
-
-&iwdg2 {
-	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
-};
-
-&mdma1 {
-	resets = <&scmi_reset RST_SCMI_MDMA>;
-};
-
-&mlahb {
-	resets = <&scmi_reset RST_SCMI_MCU>;
-};
-
-&rcc {
-	compatible = "st,stm32mp1-rcc-secure", "syscon";
-	clock-names = "hse", "hsi", "csi", "lse", "lsi";
-	clocks = <&scmi_clk CK_SCMI_HSE>,
-		 <&scmi_clk CK_SCMI_HSI>,
-		 <&scmi_clk CK_SCMI_CSI>,
-		 <&scmi_clk CK_SCMI_LSE>,
-		 <&scmi_clk CK_SCMI_LSI>;
-};
-
-&rng1 {
-	clocks = <&scmi_clk CK_SCMI_RNG1>;
-	resets = <&scmi_reset RST_SCMI_RNG1>;
-};
-
-&rtc {
-	clocks = <&scmi_clk CK_SCMI_RTCAPB>, <&scmi_clk CK_SCMI_RTC>;
-};
